tests/test_multidict.py

```diff
@@ -823,3 +823,14 @@ def test_keys__repr__(self, cls: Type[CIMultiDict[str]]) -> None:
     def test_values__repr__(self, cls: Type[CIMultiDict[str]]) -> None:
         d = cls([("KEY", "value1")], key="value2")
         assert repr(d.values()) == "_ValuesView('value1', 'value2')"
+
+    @pytest.mark.xfail(reason="CI sets are not implemented yet")
+    def test_keys_case_insensitive_union(
+        self,
+        cls: Type[CIMultiDict[str]],
+    ) -> None:
+        d = cls([("KEY", "one"),])
+
+        assert "key" in d.keys()
+        assert d.keys().isdisjoint({"key"})
+        assert {"key"} & d.keys() == {"KEY"}
```

Comment:
I think it may be safer to return the values from the set, rather than the keys:
```suggestion
        assert {"key"} & d.keys() == {"key"}
```

This makes it predictable what the exact values will be in the set, if a developer wants to get them afterwards (e.g. it means they are normalised to whatever the developer is looking for).

Reply:
Good suggestion, thanks!